pluton/spawn/containercache/sshcache.go

```diff
@@ -0,0 +1,278 @@
+
+			elasped := time.Since(start)
+			fmt.Printf("Extracting and pulling %v took %v\n", name.Name, elasped)
+
+		default:
+			return fmt.Errorf("invalid container name Engine must either be 'rkt' or 'docker' got %v", name.Engine)
+		}
+	}
+
+	elasped := time.Since(start)
+	fmt.Printf("Total prefetch time took %v\n", elasped)
+
+	start = time.Now()
+
+	// extract containers to known location for Pull to pick up later
+	err = extract(bastion, names)
+	if err != nil {
+		return fmt.Errorf("error extracting containers: %v", err)
+	}
+
+	elasped = time.Since(start)
+	fmt.Printf("Total time extracting prefetched containers to disk: %v\n", elasped)
+
+	return nil
+}
+
+// Will add a timeout to the SSH command
+func sshWithTimeout(m platform.Machine, cmd string, timeout time.Duration) error {
+	client, err := m.SSHClient()
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	var outBuf = bytes.NewBuffer(nil)
+	var errBuf = bytes.NewBuffer(nil)
+	session.Stdout = outBuf
+	session.Stderr = errBuf
+
+	err = session.Start(cmd)
+	if err != nil {
+		return err
+	}
+
+	errc := make(chan error)
+	go func() { errc <- session.Wait() }()
+
+	select {
+	case err := <-errc:
+		if err != nil {
+			return fmt.Errorf("error prefetching cmd: %v -- %s", cmd, append(outBuf.Bytes(), errBuf.Bytes()...))
+		}
+	case <-time.After(timeout):
+		return fmt.Errorf("timeout prefetching: %v -- %s", cmd, append(outBuf.Bytes(), errBuf.Bytes()...))
+	}
+
+	return nil // unreachable
+}
+
+// Extract existing containers to fs
+func extract(bastion platform.Machine, names []ImageName) error {
+	out, err := bastion.SSH(fmt.Sprintf("mkdir %v", cacheDir))
+	if err != nil {
+		return fmt.Errorf("creating cache dir: %v -- %s", err, out)
+	}
+
+	for _, name := range names {
+		switch name.Engine {
+		case "rkt":
+			out, err = bastion.SSH(fmt.Sprintf("sudo rkt image export %v %v/%v.aci", name.Name, cacheDir, strings.Replace(name.Name, "/", ".", -1)))
+			if err != nil {
+				return fmt.Errorf("rkt image export: %v -- %s", err, out)
+			}
+		case "docker":
+			out, err = bastion.SSH(fmt.Sprintf("docker save %v -o %v/%v.tar", name.Name, cacheDir, strings.Replace(name.Name, "/", ".", -1)))
+			if err != nil {
+				return fmt.Errorf("docker save: %v -- %s", err, out)
+			}
+		default:
+			return fmt.Errorf("invalid container name Engine must either be 'rkt' or 'docker' got %v", name.Engine)
+		}
+	}
+
+	out, err = bastion.SSH("sudo chown -R core " + cacheDir)
+	if err != nil {
+		return fmt.Errorf("%v: %s", err, out)
+	}
+
+	return nil
+}
+
+func copyPublicKeys(bastion platform.Machine, machines []platform.Machine) error {
+	for _, m := range machines {
+		err := platform.TransferFile(bastion, "/home/core/.ssh/bastion.key.pub", m, "/home/core/bastion.key.pub")
+		if err != nil {
+			return err
+		}
+
+		out, err := m.SSH("cat /home/core/bastion.key.pub >> /home/core/.ssh/authorized_keys")
```

Comment:
This conflicts with anything update-ssh-keys does, instead you can do `update-ssh-keys -a bastion bastion.key.pub`